src/main/java/com/conveyal/datatools/manager/controllers/api/LabelController.java

```diff
@@ -0,0 +1,217 @@
+                    fs.labels = newLabels;
+                    Persistence.feedSources.replace(fs.id, fs);
+                }
+            });
+
+            label.delete();
+            return label;
+        } catch (Exception e) {
+            logMessageAndHalt(req, 500, "Unknown error occurred while deleting label.", e);
+            return null;
+        }
+    }
+
+    /**
+     * Helper function returns label if user has permission for specified action.
+     * @param req spark Request object from API request
+     * @param action action type (either "view" or Permission.MANAGE)
+     * @return label object for ID
+     */
+    private static Label requestLabelById(Request req, Actions action) {
+        String id = req.params("id");
+        if (id == null) {
+            logMessageAndHalt(req, 400, "Please specify id param.");
+        }
+        Label label = Persistence.labels.getById(id);
+        checkLabelPermissions(req, label, action);
+
+        return label;
+    }
+
+    /** Helper method will halt execution and throw an error message if a given
+     * label is not allowed to be accessed by the current user (taken from the req object)
+     *
+     * @param req       Spark request used for determining user permissions
+     * @param label     Label to check
+     * @param action    Action to be taken on label, which changes who can do what
+     */
+    public static void checkLabelPermissions(Request req, Label label, Actions action) {
+        Auth0UserProfile userProfile = req.attribute("user");
+        // check for null label
+        if (label == null) {
+            logMessageAndHalt(req, 400, "Label ID does not exist.");
+            return;
+        }
+
+        boolean isProjectAdmin = userProfile.canAdministerProject(label.projectId, label.organizationId());
+        switch (action) {
+            case CREATE:
+            case MANAGE:
+            case EDIT:
+                // Only project admins can edit, manage, or create labels
+                if (!isProjectAdmin) {
+                    logMessageAndHalt(req, 403, "User is not admin so cannot update or create labels.");
+                }
+                break;
+            case VIEW:
+                if (label.adminOnly && !isProjectAdmin) {
+                    logMessageAndHalt(req, 403, "User is not admin so cannot view admin-only label.");
+                }
+                break;
+            default:
+                // Incorrect query, so fail
+                logMessageAndHalt(req, 400, "Not enough information supplied to determine label access.");
+                break;
+        }
+
+        // If we make it here, user has permission and the requested label is valid
+    }
+
+    public static void register (String apiPrefix) {
+        get(apiPrefix + "secure/label/:id", LabelController::getLabel, json::write);
+        get(apiPrefix + "secure/label", LabelController::getAllLabels, json::write);
```

Comment:
Is this endpoint still needed?

Reply:
It isn't presently used by the UI, but I thought it might be worth keeping around to make future work easier? But do not have strong opinion on that so happy to remove if you think it's a better idea to remove it

Reply:
In general I think it's good to remove unused code, but I'll let the final reviewer have the final say.